src/utils/columnPinning.ts

```diff
@@ -0,0 +1,29 @@
+import type React from 'react';
+
+import type {NoStrictEntityMods} from '@bem-react/classname';
+import type {Column} from '@tanstack/react-table';
+
+export function getCommonPinningStyles<TData>(column: Column<TData>): React.CSSProperties {
+    const isPinned = column.getIsPinned();
+
+    return {
+        left: isPinned === 'left' ? `${column.getStart('left')}px` : undefined,
+        right: isPinned === 'right' ? `${column.getAfter('right')}px` : undefined,
+        position: isPinned ? 'sticky' : 'relative',
+        width: column.getSize(),
+        zIndex: isPinned ? 1 : 0,
+    };
+}
+
+export function getCommonPinningClassModes<TData>(column: Column<TData>): NoStrictEntityMods {
```

Comment:
Let's rename `getCommonPinningClassModes` to `getCellClassModes` and set common class-modes here (not just pinning-related).

Reply:
I separated `getCommonPinningClassModes` into 2 different functions. First for default cell - `getCellClassModes `, and the second for header cells - `getHeaderCellClassModes `. I did this because it has different logic for default cell and header cell, and i think it's not correct to mix them.

And I also made another function for pinned columns class modes to reuse it in functions for cells.